examples/utils_data.jl

```diff
@@ -428,46 +428,48 @@ function get_data_correlated_ou2d_timeseries(batchsize;
                                  
     @assert xor(read, save)
 
+    # TODO: this is an extremly memory inefficient way of creating the dataloaders
     rawtrain = CliMADatasets.CorrelatedOU2D(:train; f = f, resolution=resolution, Tx=FT);
     samp_per_τ = Int(round(rawtrain.metadata["τ"]/ rawtrain.metadata["dt_save"]))
     rawtrain = rawtrain.features
     rawtest = CliMADatasets.CorrelatedOU2D(:test; f = f, resolution=resolution, Tx=FT)[:]
 
     # Create train and test datasets
+    stride = Int(samp_per_τ)
+    @assert stride > ntime
 
-    @assert pairs_per_τ <= samp_per_τ/2
-    stride = Int(floor(samp_per_τ/pairs_per_τ))
-    
     nsamp_train = size(rawtrain)[end]
-    npairs_train = length(Array(1:stride:(nsamp_train-1)))
-    xtrain  = zeros(FT, (resolution, resolution, 2, npairs_train))
-    xtrain[:,:,1,:] .= rawtrain[:,:,1:stride:(nsamp_train-1)]
-    xtrain[:,:,2,:] .= rawtrain[:,:,2:stride:nsamp_train]
+    nseries_train = div(nsamp_train, stride) 
+    xtrain  = zeros(FT, (resolution, resolution, ntime, 1, nseries_train))
```

Comment:
hardcodes nchannels, but I think this is fine since this is particular to this dataset.